taca/element/Element_Runs.py

```diff
@@ -0,0 +1,1270 @@
+        dest = os.path.join(metadata_archive, self.NGI_run_id)
+        if not os.path.exists(dest):
+            os.makedirs(dest)
+        for f in files_to_copy: # UnassignedSequences.csv missing in NoIndex case
+            if os.path.exists(f):
+                shutil.copy(f, dest)
+            else:
+                logger.warning(f"File {f} missing for run {self.run}")
+
+    def make_transfer_indicator(self):
+        transfer_indicator = os.path.join(self.run_dir, ".rsync_ongoing")
+        Path(transfer_indicator).touch()
+
+    def transfer(self):
+        transfer_details = self.CONFIG.get("element_analysis").get("transfer_details")
+        command = (
+            "rsync"
+            + " -rLav"
+            + f" --chown={transfer_details.get('owner')}"
+            + f" --chmod={transfer_details.get('permissions')}"
+            + " --exclude BaseCalls"
+            + " --exclude Alignment"
+            + f" {self.run_dir}"
+            + f" {transfer_details.get('user')}@{transfer_details.get('host')}:/aviti"
+            + f"; echo $? > {os.path.join(self.run_dir, '.rsync_exit_status')}"
+        )
+        try:
+            p_handle = subprocess.Popen(command, stdout=subprocess.PIPE, shell=True)
+            logger.info(
+                "Transfer to analysis cluster "
+                f"started for run {self} on {datetime.now()}"
+                f"with p_handle {p_handle}"
+            )
+        except subprocess.CalledProcessError:
+            logger.warning(
+                "An error occurred while starting transfer to analysis cluster "
+                f"for {self} on {datetime.now()}."
+            )
+        return
+
+    def remove_transfer_indicator(self):
+        transfer_indicator = os.path.join(self.run_dir, ".rsync_ongoing")
+        Path(transfer_indicator).unlink()
+
+    def update_transfer_log(self):
+        """Update transfer log with run id and date."""
+        try:
+            with open(self.transfer_file, "a") as f:
+                tsv_writer = csv.writer(f, delimiter="\t")
+                tsv_writer.writerow([self.NGI_run_id, str(datetime.now())])
+        except OSError:
+            msg = f"{self}: Could not update the transfer logfile {self.transfer_file}"
+            logger.error(msg)
+            raise OSError(msg)
+
+    def update_paths_after_archiving(self, new_location):
+        self.run_dir = os.path.join(
+            new_location, self.NGI_run_id
+        )  # Needs to be redirected to new location so that TACA can find files to upload to statusdb
+        self.run_parameters_file = os.path.join(self.run_dir, "RunParameters.json")
+        self.run_stats_file = os.path.join(self.run_dir, "AvitiRunStats.json")
+        self.run_manifest_file_from_instrument = os.path.join(
+            self.run_dir, "RunManifest.json"
+        )
+        self.run_uploaded_file = os.path.join(self.run_dir, "RunUploaded.json")
+
+    def archive(self):
```

Comment:
I think this is a bit misleadingly name. Archiving normally means sending to pdc right?

Reply:
Renamed 👍